ubuntu-installation.md

```diff
@@ -0,0 +1,136 @@
+
+
+### Project Page
+
+The Github project page is at [https://github.com/rubyforgood/partner](https://github.com/rubyforgood/partner).
+
+
+### Installing OS Packages
+
+You'll need to add some packages to those that are included in the installation by default:
+
+`sudo apt install curl  git  nodejs  npm  postgresql  libpq-dev`
+
+
+### Installing the Partner Software
+
+If you will be using this machine for development, you will need to `git clone` the repo:
+
+`git clone https://github.com/rubyforgood/partner.git`
+
+If you only need to have the files but don't need to use git, you can download the zip file and unzip it:
+
+```
+curl -o partner.zip https://codeload.github.com/rubyforgood/partner/zip/master \
+    && unzip partner.zip \
+    && rm partner.zip
+```
+
+
+### Installing Yarn
+ 
+Go to your partner project directory and do this:
+
+```
+sudo npm -g install yarn
+yarn install
+```
+
+
+### Installing rvm
+
+Go to https://rvm.io/rvm/install and find and run the commands that install the gpg key and install rvm. At the time of this writing (2019-07-26), they are:
+
+```
+gpg --keyserver hkp://pool.sks-keyservers.net --recv-keys 409B6B1796C275462A1703113804BB82D39DC0E3 7D2BAF1CF37B13E2069D6956105BD0E739499BDB
+\curl -sSL https://get.rvm.io | bash
+```
+
+_Open a new terminal tab or window (this will not work in your current shell!)._ To test rvm, do:
+
+`rvm list`
+
+If you see something like ‘rvm not found’, you may need to change your terminal settings so that something like “Run command as a login shell” is selected. In the default Ubuntu terminal, you would go to the Edit menu, then select the profile to edit, then select the “Command” tab, and select “Run command as a login shell”. Then open a new shell to proceed.
+
+
+### Installing Ruby and Ruby Gems
+
+Install the required version of Ruby (you may be prompted for your system password so that rvm can install some required packages):
+
+`rvm install $(cat .ruby-version)`
+
+Then, from your project directory, install the gems required for the project:
+
+`bundle`
+
+
+### Configuring Postgres
+
+By default, Postgres has no password for the admin user. You'll need to set one. Run `psql` as the Unix user `postgres` that was added by the Postgres installation:
+
+`sudo -u postgres psql`
+
+In `psql`, the following command would add a password whose value is `password`:
+
+`alter user postgres password 'password';`
+
+Don't forget that semicolon!
+
+If the command succeeds, `psql` will respond with:
+
+`ALTER ROLE`
+
+You can now exit `psql`, by pressing `<Ctrl-D>` or typing `\q<Enter>`.
+
+
+### The .env File
+
+This project uses a `.env` file in the project root directory, and it is recommended to use it to include a) the Postgres credentials, and b) the Diaperbank key and endpoint (needed for running the tests locally).
+
+In the project root directory, create an `.env` file containing:
+
+```
+export PG_USERNAME=postgres
```

Comment:
I dont think these two need `export` in front of them.

Reply:
Sorry, yes, you're right. I've fixed this.